spec/graphql/language/parser_spec.rb

```diff
@@ -304,6 +310,136 @@
+        GraphQL.parse("{ \xeffield }")
+      end
+
+      assert_includes(e.message, "Parse error on \"\\xEF\"")
+    end
+
+    it "rejects vertical tabs" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ \vfield }")
+      end
+
+      assert_includes(e.message, "Parse error on \"\\v\"")
+    end
+
+    it "rejects form feed" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ \ffield }")
+      end
+
+      assert_includes(e.message, "Parse error on \"\\f\"")
+    end
+
+    it "rejects no break space" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ \xa0field }")
+      end
+
+      assert_includes(e.message, "Parse error on \"\\xA0\"")
+    end
+
+    it "rejects unterminated strings" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("\"")
+      end
+
+      assert_includes(e.message, "Parse error on \"\\\"\"")
+
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("\"\n\"")
+      end
+
+      assert_includes(e.message, "Parse error on \"\\n\"")
+    end
+
+    it "rejects bad escape sequence in strings" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ field(arg:\"\\x\") }")
+      end
+
+      assert_includes(e.message, "Parse error on bad Unicode escape sequence")
+    end
+
+    it "rejects incomplete escape sequence in strings" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ field(arg:\"\\u1\") }")
+      end
+
+      assert_includes(e.message, "bad Unicode escape sequence")
+    end
+
+    it "rejects unicode escape with bad chars" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ field(arg:\"\\u0XX1\") }")
+      end
+
+      assert_includes(e.message, "bad Unicode escape sequence")
+
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ field(arg:\"\\uXXXX\") }")
+      end
+
+      assert_includes(e.message, "bad Unicode escape sequence")
+
+
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ field(arg:\"\\uFXXX\") }")
+      end
+
+      assert_includes(e.message, "bad Unicode escape sequence")
+
+
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ field(arg:\"\\uXXXF\") }")
+      end
+
+      assert_includes(e.message, "bad Unicode escape sequence")
+    end
+
+    it "rejects fragments named 'on'" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("fragment on on on { on }")
+      end
+
+      assert_includes(e.message, "Parse error on \"on\"")
+    end
+
+    it "rejects fragment spread of 'on'" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ ...on }")
+      end
+
+      assert_includes(e.message, "Parse error on \"}\"")
+    end
+
+    it "rejects null value" do
+      e = assert_raises(GraphQL::ParseError) do
+        GraphQL.parse("{ fieldWithNullableStringInput(input: null) }")
```

Comment:
Is it _legal_ GraphQL to have an enum value named `null`?  Ok, I looked, apparently not: 

![image](https://cloud.githubusercontent.com/assets/2231765/15166980/7430d4fe-16f2-11e6-8798-7f5a39f8d2fd.png)


:+1: